### ISSUE TITLE
Failed to connect to WM

### ISSUE THREAD

**Issue by andersfylling** [pierrechevalier83/workstyle#28]
I logged out of my session after I installed this project and logged back in to start sway, but it seems my workspaces aren't renamed (I'm only seeing the numbers, same as the default sway behavior).

.zprofile:
```sh
# https://intellij-support.jetbrains.com/hc/en-us/community/posts/4402682513426-Unable-to-install-IntelliJ-in-Arch-Linux-Sway?page=1#community_comment_4402781934354
export _JAVA_AWT_WM_NONREPARENTING=1

if [ -z $DISPLAY ] && [ "$(tty)" = "/dev/tty1" ]; then
	sh .config/sway/start.sh
fi
```
start.sh:
```sh
LOG_FILE_PATH="/var/log/sway/$(date +'%Y-%m-%d').log"
sway -d -V >> ${LOG_FILE_PATH} 2>&1
```

added to my sway config:
```
exec_always --no-startup-id workstyle &> /tmp/workstyle.log
```

tail workstyle.log:
```
[2022-02-20T03:09:12Z ERROR workstyle] Couldn't connect to WM
[2022-02-20T03:09:12Z ERROR workstyle] Failed to connect to WM. Will try again in 1 second
[2022-02-20T03:09:13Z ERROR workstyle] Couldn't connect to WM
[2022-02-20T03:09:13Z ERROR workstyle] Failed to connect to WM. Will try again in 1 second
[2022-02-20T03:09:14Z ERROR workstyle] Couldn't connect to WM
[2022-02-20T03:09:14Z ERROR workstyle] Failed to connect to WM. Will try again in 1 second
[2022-02-20T03:09:15Z ERROR workstyle] Couldn't connect to WM
[2022-02-20T03:09:15Z ERROR workstyle] Failed to connect to WM. Will try again in 1 second
[2022-02-20T03:09:16Z ERROR workstyle] Couldn't connect to WM
[2022-02-20T03:09:16Z ERROR workstyle] Failed to connect to WM. Will try again in 1 second
```

system info:
```
arch linux
5.15.24-1-lts
```

**Comment by MaxVerevkin** [comment]
Did version 0.7 work without issues?

**Comment by pierrechevalier83** [comment]
> I logged out of my session after I installed this project and logged back in to start sway, but it seems my workspaces aren't renamed (I'm only seeing the numbers, same as the default sway behavior).
> 
> .zprofile:
> 
> ```shell
> # https://intellij-support.jetbrains.com/hc/en-us/community/posts/4402682513426-Unable-to-install-IntelliJ-in-Arch-Linux-Sway?page=1#community_comment_4402781934354
> export _JAVA_AWT_WM_NONREPARENTING=1
> 
> if [ -z $DISPLAY ] && [ "$(tty)" = "/dev/tty1" ]; then
> 	sh .config/sway/start.sh
> fi
> ```
> 
> added to my sway config:
> 
> ```
> exec_always --no-startup-id workstyle &> /tmp/workstyle.log
> ```
> 
> tail workstyle.log:
> 
> ```
> [2022-02-20T03:09:12Z ERROR workstyle] Couldn't connect to WM
> [2022-02-20T03:09:12Z ERROR workstyle] Failed to connect to WM. Will try again in 1 second
> [2022-02-20T03:09:13Z ERROR workstyle] Couldn't connect to WM
> [2022-02-20T03:09:13Z ERROR workstyle] Failed to connect to WM. Will try again in 1 second
> [2022-02-20T03:09:14Z ERROR workstyle] Couldn't connect to WM
> [2022-02-20T03:09:14Z ERROR workstyle] Failed to connect to WM. Will try again in 1 second
> [2022-02-20T03:09:15Z ERROR workstyle] Couldn't connect to WM
> [2022-02-20T03:09:15Z ERROR workstyle] Failed to connect to WM. Will try again in 1 second
> [2022-02-20T03:09:16Z ERROR workstyle] Couldn't connect to WM
> [2022-02-20T03:09:16Z ERROR workstyle] Failed to connect to WM. Will try again in 1 second
> ```
> 
> system info:
> 
> ```
> arch linux
> 5.15.24-1-lts
> ```

Could you share the contents of your `~/.config/sway/start.sh` to allow me to try and reproduce the issue locally?
I'm starting sway through gdm on my machine and haven't observed this behaviour.

**Comment by pierrechevalier83** [comment]
I've also added some logging in this PR: https://github.com/pierrechevalier83/workstyle/pull/29

May I ask you to update workstyle (btw, how did you install it? from the AUR?) and share your logs again?

**Comment by pierrechevalier83** [comment]
> I logged out of my session after I installed this project and logged back in to start sway, but it seems my workspaces aren't renamed (I'm only seeing the numbers, same as the default sway behavior).

Actually, if it's the first time you're trying `workstyle`, could you try to run `RUST_LOG=debug workstyle` manually from a CLI on `sway` and see what happens? That might help rule out any problems with configuration scripts not doing what you think they're doing and isolate any problems caused by the `worskstlyle` application itself.

**Comment by andersfylling** [comment]
I'm not certain why it happened. It's not deterministic, and works most of the time now. Haven't experienced this since I opened the issue.

If I come across this again I'll see if I can share the rust logs.


start.sh:
```sh
LOG_FILE_PATH="/var/log/sway/$(date +'%Y-%m-%d').log"
sway -d -V >> ${LOG_FILE_PATH} 2>&1
```

**Comment by pierrechevalier83** [comment]
> I used the cargo command from the README.md file.

In that case, I'll need to publish the crate again for you to get the extra logging.

**Comment by pierrechevalier83** [comment]
I published 0.8.2 to crates.io, now with some more logging. I've also added documentation to the README on installing the latest git head from the AUR in case that's of interest.

@andersfylling , if you could update `workstyle` with either method, that may help with getting better logs if you hit this issue again.

**Comment by pierrechevalier83** [comment]
@andersfylling , I'm going to go ahead and close this issue now as there hasn't been an update on it for a while and the rootcause may have been fixed since the version that was giving you trouble. Please, don't hesitate to re-open it if you hit it again.
